sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -935,4 +935,106 @@ abstract class SchemaPruningSuite
       .count()
     assert(count == 0)
   }
+
+  testSchemaPruning("SPARK-38977: schema pruning with correlated EXISTS subquery") {
```

Comment:
All of these queries would previously fail.

Reply:
this bug only happens for v2 tables, not file source tables?

Reply:
I guess it will fail for both as the same method is used. Tests cover V1 and V2 so it should work for both now.